ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/PeerSimulator/Handlers.hs

```diff
@@ -50,20 +55,19 @@ handlerFindIntersection ::
   IOLike m =>
   StrictTVar m (Point TestBlock) ->
   BlockTree TestBlock ->
-  AdvertisedPoints ->
   [Point TestBlock] ->
-  STM m (FindIntersect, [String])
-handlerFindIntersection currentIntersection blockTree points clientPoints = do
+  AdvertisedPoints ->
+  STM m (Maybe FindIntersect, [String])
+handlerFindIntersection currentIntersection blockTree clientPoints points = do
   let TipPoint tip' = tip points
       tipPoint = getTipPoint tip'
       fragment = fromJust $ BT.findFragment tipPoint blockTree
   case intersectWith fragment clientPoints of
     Nothing ->
-      pure (IntersectNotFound tip', [])
+      pure (Just (IntersectNotFound tip'), [])
     Just intersection -> do
       writeTVar currentIntersection intersection
-      pure (IntersectFound intersection tip', [])
-  where
+      pure (Just (IntersectFound intersection tip'), [])
```

Comment:
What's the reason to return a `Maybe FindIntersect` if we always return a `Just`? Or did you mean to return `Nothing` when `BT.findFragment` returns `Nothing`?

Reply:
because it's the unified API of `runHandler`